lib/utils/index.ts

```diff
@@ -0,0 +1,21 @@
+
+type StrategyOptions = {
+  onError?: 'throw' | 'reject' | 'ignore' | 'ignoreAsync';
+};
+
+export function raiseStrategy(options: StrategyOptions, defaultStrategy: string) {
```

Comment:
`defaultStrategy` violates the SRP. It's the responsibility of the caller to provide the default.